src/lib/utils.ts

```diff
@@ -282,23 +283,44 @@ export function encodeKey(
     case 'mappingwithgrouping':
     case 'singleton':
     case 'mapping':
+      if (schema.valueType.includes('[CompactBytesArray]')) {
+        const valueType = schema.valueType.replace('[CompactBytesArray]', '');
+        const valueContent = schema.valueContent.replace(
+          '[CompactBytesArray]',
+          '',
+        );
+
+        if (
+          Array.isArray(value[0]) &&
+          valueType[0] === '(' &&
+          valueType[valueType.length - 1] === ')'
+        ) {
+          const encodedTuples = (value as string[][]).map((element) => {
```

Comment:
whenever you write `as xxx` you are "forcing" TypeScript to cast a value to a certain type.
Very likely, you are doing something wrong or not the best way.

Adding these "as" makes the TypeScript types less relevant and more fragile.

Here, at this stage, TS thinks this value is:
![image](https://user-images.githubusercontent.com/477945/214849360-4fd31d37-7d96-472f-baa9-01c04324b0d8.png)

You can see in the screenshot that something is probably off.
TypeScript thinks this variable can be of multiple types, but you force it to be one of them.

The best way to have TS be sure that the type is string[][], is to do the propre __JavaScript__ checks.